plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -573,6 +591,25 @@ private static void verifySupportedDate(long value)
         }
     }
 
+    private static LongWriteFunction timestampSecondsWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long epochSecond = floorDiv(value, MICROSECONDS_PER_SECOND);
+            int nanoFraction = floorMod(value, MICROSECONDS_PER_SECOND) * NANOSECONDS_PER_MICROSECOND;
+            verify(nanoFraction == 0, "Nanos of second must be zero: '%s'", value);
+            verifySupportedTimestamp(epochSecond);
+            statement.setObject(index, LocalDateTime.ofEpochSecond(epochSecond, 0, UTC));
+        };
+    }
+
+    private static void verifySupportedTimestamp(long epochSecond)
+    {
+        if (epochSecond < MIN_SUPPORTED_TIMESTAMP_EPOCH || epochSecond > MAX_SUPPORTED_TIMESTAMP_EPOCH) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss");
+            throw new TrinoException(INVALID_ARGUMENTS, format("Timestamp must be between %s and %s: %s", MIN_SUPPORTED_TIMESTAMP.format(formatter), MAX_SUPPORTED_TIMESTAMP.format(formatter), LocalDateTime.ofEpochSecond(epochSecond, 0, UTC).format(formatter)));
```

Comment:
If we remove validation on our side does ClickHouse return a useful error?

Or let's make the error message convey that ClickHouse supports timestamps in the specified range. Current message looks like Trino limitation.

Reply:
ClickHouse stores incorrect results for invalid timestamps. https://github.com/trinodb/trino/blob/master/plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java#L448